interfaces/tests/file_int/file_int.c

```diff
@@ -0,0 +1,559 @@
+    write_context.post_callback_value = 42;
+
+    unsigned char destination[5];
+    READ_COMPLETE_CONTEXT read_context;
+    read_context.pre_callback_value = 43;
+    interlocked_exchange(&read_context.value, read_context.pre_callback_value);
+    read_context.post_callback_value = 44;
+
+    uint32_t read_position = 2;
+
+    char filename[] = "read_across_eof_fails.txt";
+    ASSERT_IS_FALSE(check_file_exists(filename));
+    EXECUTION_ENGINE_HANDLE execution_engine = execution_engine_create(NULL);
+    FILE_HANDLE file_handle = file_create(execution_engine, filename, NULL, NULL);
+
+    ///act
+    FILE_WRITE_ASYNC_RESULT write_result = file_write_async(file_handle, source, size, 0, write_callback, &write_context);
+    wait_on_address_helper(&write_context.value, &write_context.pre_callback_value, UINT32_MAX);
+
+    ///assert
+    ASSERT_ARE_EQUAL(FILE_WRITE_ASYNC_RESULT, FILE_WRITE_ASYNC_OK, write_result);
+    ASSERT_ARE_EQUAL(int32_t, write_context.post_callback_value, interlocked_or(&write_context.value, 0), "value should be post_callback_value");
+    ASSERT_IS_TRUE(write_context.did_write_succeed);
+
+    ///act
+    file_read_async(file_handle, destination, sizeof(destination), read_position, read_callback, &read_context);
+    wait_on_address_helper(&read_context.value, &read_context.pre_callback_value, UINT32_MAX);
+
+    ///assert
+    ASSERT_ARE_EQUAL(int32_t, read_context.post_callback_value, interlocked_or(&read_context.value, 0), "value should be post_callback_value");
+    ASSERT_IS_FALSE(read_context.did_read_succeed);
+
+    //cleanup
+    file_destroy(file_handle);
+}
+
+/*Tests_SRS_FILE_43_039: [ If position + size exceeds the size of the file, user_callback shall be called with success as false. ]*/
+TEST_FUNCTION(read_beyond_eof_fails)
+{
+    ///arrange
+    const int size = 5;
+
+    unsigned char source[] = "abcd";
+    WRITE_COMPLETE_CONTEXT write_context;
+    write_context.pre_callback_value = 41;
+    interlocked_exchange(&write_context.value, write_context.pre_callback_value);
+    write_context.post_callback_value = 42;
+
+    unsigned char destination[5];
+    READ_COMPLETE_CONTEXT read_context;
+    read_context.pre_callback_value = 43;
+    interlocked_exchange(&read_context.value, read_context.pre_callback_value);
+    read_context.post_callback_value = 44;
+
+    uint32_t read_position = 5;
+
+    char filename[] = "read_beyond_eof_fails.txt";
+    ASSERT_IS_FALSE(check_file_exists(filename));
+    EXECUTION_ENGINE_HANDLE execution_engine = execution_engine_create(NULL);
+    FILE_HANDLE file_handle = file_create(execution_engine, "read_beyond_eof_fails.txt", NULL, NULL);
+
+    ///act
+    FILE_WRITE_ASYNC_RESULT write_result = file_write_async(file_handle, source, size, 0, write_callback, &write_context);
+    wait_on_address_helper(&write_context.value, &write_context.pre_callback_value, UINT32_MAX);
+    
+    ///assert

```

Comment:
>///assert [](start = 4, length = 9)

nit: really if you are testing "read beyond eof fails" then the write part is all "arrange" and the "act"/"assert" happens on lines 472 and 476. Could remove these comments (line 463, 467)